src/runtime_client.rs

```diff
@@ -0,0 +1,531 @@
+        )?;
+        obj.set(
+            "getRemainingTimeInMillis",
+            self.get_remaining_time_in_millis,
+        )?;
+        obj.set("clientContext", self.client_context)?;
+        obj.set("cognitoIdentityJson", self.cognito_identity_json)?;
+        Ok(obj.into_value())
+    }
+}
+
+#[derive(Clone)]
+struct LambdaEnvironment {
+    pub log_group_name: String,
+    pub log_stream_name: String,
+    pub function_name: String,
+    pub function_version: String,
+    pub memory_limit_in_mb: usize,
+}
+
+impl LambdaEnvironment {
+    fn new() -> Self {
+        Self {
+            log_group_name: env::var(ENV_AWS_LAMBDA_LOG_GROUP_NAME).unwrap_or_default(),
+            log_stream_name: env::var(ENV_AWS_LAMBDA_LOG_STREAM_NAME).unwrap_or_default(),
+            function_name: env::var(ENV_AWS_LAMBDA_FUNCTION_NAME).unwrap_or_default(),
+            function_version: env::var(ENV_AWS_LAMBDA_FUNCTION_VERSION).unwrap_or_default(),
+            memory_limit_in_mb: env::var(ENV_AWS_LAMBDA_FUNCTION_MEMORY_SIZE)
+                .unwrap_or("128".into())
+                .parse()
+                .unwrap_or_default(),
+        }
+    }
+}
+
+struct NextInvocationResponse<'js, 'a> {
+    event: Value<'js>,
+    context: LambdaContext<'js, 'a>,
+}
+
+struct RuntimeConfig {
+    runtime_api: String,
+    handler: String,
+    iterations: usize,
+}
+
+impl RuntimeConfig {
+    fn default(ctx: &Ctx) -> Result<Self> {
+        Ok(Self {
+            runtime_api: env::var(AWS_LAMBDA_RUNTIME_API).map_err(|_| {
+                Exception::throw_message(
+                    ctx,
+                    &format!(
+                        "Environment variable {} is not defined.",
+                        AWS_LAMBDA_RUNTIME_API
+                    ),
+                )
+            })?,
+            handler: env::var(ENV_LAMBDA_HANDLER)
+                .or_else(|_| env::var(ENV_UNDERSCORE_HANDLER))
+                .map_err(|_| {
+                    Exception::throw_message(
+                        ctx,
+                        &format!(
+                            "Environment {} or {} is not defined.",
+                            ENV_UNDERSCORE_HANDLER, ENV_LAMBDA_HANDLER
+                        ),
+                    )
+                })?,
+            iterations: env::var(ENV_UNDERSCORE_EXIT_ITERATIONS)
+                .ok()
+                .and_then(|i| i.parse().ok())
+                .unwrap_or_default(),
+        })
+    }
+}
+
+pub async fn start(ctx: &Ctx<'_>) -> Result<()> {
+    start_with_cfg(ctx, RuntimeConfig::default(ctx)?).await
+}
+
+async fn start_with_cfg(ctx: &Ctx<'_>, config: RuntimeConfig) -> Result<()> {
+    let (module_name, handler_name) = get_module_and_handler_name(ctx, &config.handler)?;
+    let task_root = get_task_root();
+
+    let js_handler_module: Object = Module::import(ctx, format!("{}/{}", task_root, module_name))?;
+    let js_init = js_handler_module.get::<_, Value>("init")?;
```

Comment:
This `init` support was a legacy feature, and no longer supported in the latest node.js runtimes. 

Is there any particular reasons to keep it in LLRT? If not, I recommend dropping it.

Reply:
I want to keep it until top-level await is there, otherwise users can't do async tasks pre-init. Timeline for this is 1-2 months depending on a significant downstream engine upgrade in `rquickjs`